lib/auth/integration/integrationv1/awsoidc.go

```diff
@@ -897,3 +898,75 @@ func (s *AWSOIDCService) Ping(ctx context.Context, req *integrationpb.PingReques
 		UserId:    resp.UserID,
 	}, nil
 }
+
+func (s *Service) deleteAWSOIDCAssociatedResources(ctx context.Context, authCtx *authz.Context, ig types.Integration) error {
+	if err := authCtx.CheckAccessToKind(types.KindDiscoveryConfig,
+		types.VerbList, types.VerbUpdate, types.VerbDelete); err != nil {
+		return trace.AccessDenied(
+			`deleting AWS OIDC integration requires 'list', 'update', and 'delete' ` +
+				`permissions for 'discovery_config' kind to remove associated resources`)
+	}
+
+	if err := authCtx.CheckAccessToKind(types.KindAppServer,
+		types.VerbList, types.VerbDelete); err != nil {
+		return trace.AccessDenied(
+			`deleting AWS OIDC integration requires 'list' and 'delete' ` +
+				`permissions for 'app_server' kind to remove associated resources`)
+	}
+
+	// Remove AWS matchers from DiscoveryConfigs that reference this integration
+	for config, err := range clientutils.Resources(ctx, s.backend.ListDiscoveryConfigs) {
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		// Filter out AWSMatchers that reference integration
+		var filteredAWSMatchers []types.AWSMatcher
+		for _, awsMatcher := range config.Spec.AWS {
+			if awsMatcher.Integration == ig.GetName() {
+			} else {
+				filteredAWSMatchers = append(filteredAWSMatchers, awsMatcher)
+			}
+		}
+
+		// Delete discovery_config if no matchers exist
+		if len(filteredAWSMatchers) == 0 &&
+			len(config.Spec.Azure) == 0 &&
+			len(config.Spec.GCP) == 0 &&
+			len(config.Spec.Kube) == 0 &&
+			config.Spec.AccessGraph == nil {
+			if err := s.backend.DeleteDiscoveryConfig(ctx, config.GetName()); err != nil {
+				return trace.Wrap(err)
+			}
+
+			continue
+		}
+
+		// Update discovery_config if any AWSMatchers were removed
+		if len(filteredAWSMatchers) != len(config.Spec.AWS) {
+			config.Spec.AWS = filteredAWSMatchers
+			if _, err := s.backend.UpdateDiscoveryConfig(ctx, config); err != nil {
+				return trace.Wrap(err)
+			}
+		}
+	}
+
+	// Delete AWS access app_server associated with this integration
+	appServers, err := s.backend.GetApplicationServers(ctx, "default")
```

Comment:
What's "default"? If this is namespace, please use `defaults.Namespace` constant.